src/OperationProcessor.ts

```diff
@@ -113,33 +151,56 @@ function earlier (ts1: OperationTimestamp, ts2: OperationTimestamp): boolean {
 }
 
 /**
- * Information about a write operation relevant for the DID cache, a subset of the properties exposed by
- * WriteOperation.
+ * Information about a write operation relevant for maintaining OperationProcessor state.
  */
 interface OperationInfo {
   readonly batchFileHash: string;
   readonly type: OperationType;
   readonly timestamp: OperationTimestamp;
+  readonly parent?: OperationHash;
+
+  status: OperationStatus;
+
+  // Most recent missing ancestor if one of the ancestors is missing. Defined only if status
+  // is Unvalidated.
+  missingAncestor?: VersionId;
 }
 
 /**
- * The current implementation is a main-memory implementation without any persistence. This
+ * The current implementation of OperationProcessor is a main-memory implementation without any persistence. This
  * means that when a node is powered down and restarted DID operations need to be applied
  * from the beginning of time. This implementation will be extended in the future to support
  * persistence.
  */
-class DidCacheImpl implements DidCache {
-  /**
-   * Map a versionId to the next versionId whenever one exists.
-   */
-  private nextVersion: Map<VersionId, VersionId> = new Map();
+class OperationProcessorImpl implements OperationProcessor {
 
   /**
    * Map a operation hash to the OperationInfo which contains sufficient
    * information to reconstruct the operation.
    */
   private opHashToInfo: Map<OperationHash, OperationInfo> = new Map();
 
+  /**
+   * Map a valid versionId to the next valid versionId whenever one exists. Due to validity checks,
+   * (condition 3 in comment above Valid), the next valid version is uniquely defined if it exists.
```

Comment:
_...he next valid version is uniquely defined if it exists._ -> Can you reword? I don't get the statement.

Reply:
Tried to reword for clarity. Please check.